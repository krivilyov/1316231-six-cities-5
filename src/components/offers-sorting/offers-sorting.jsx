import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import withToggle from "../../hocs/with-toggle/with-toggle";
import {SortingTypes, SortingTitles} from '../../const';

const OffersSorting = (props) => {
  const {toggle, onOptionClick, onToggleClick, sortingType} = props;

  const handleOptionClick = (value) => {
    onOptionClick(value);
    onToggleClick();
  };

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span
        className="places__sorting-type"
        tabIndex="0"
        onClick={onToggleClick}
      >
        {SortingTitles[sortingType]}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul className={`places__options places__options--custom ${toggle ? `places__options--opened` : ``}`}>
        {SortingTypes.map((item) => (
          <li
            key={item}
            className={`places__option ${item === sortingType ? `places__option--active` : ``}`}
            tabIndex="0"
            onClick={() => handleOptionClick(item)}
          >
            {SortingTitles[item]}
          </li>
        ))}
      </ul>
      <select
        className="places__sorting-type visually-hidden"
        id="places-sorting"
        value={sortingType}
        onChange={() => {}}
      >
        {SortingTypes.map((item) => (
          <option
            key={item}
            className={`places__option ${item}`}
            value={item}
          >
            {SortingTitles[item]}
          </option>
        ))}
      </select>
    </form>
  );
};

OffersSorting.propTypes = {
  toggle: PropTypes.bool.isRequired,
  onOptionClick: PropTypes.func.isRequired,
  onToggleClick: PropTypes.func.isRequired,
  sortingType: PropTypes.string.isRequired,
};


const mapStateToProps = (state) => ({
  sortingType: state.COMMON.sortingType
});

export {OffersSorting};
export default connect(mapStateToProps)(withToggle(OffersSorting));
