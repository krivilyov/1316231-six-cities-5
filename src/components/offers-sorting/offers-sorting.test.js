import React from "react";
import renderer from "react-test-renderer";
import {OffersSorting} from "./offers-sorting";

const noop = () => {};
describe(`OffersSorting should render correctly`, () => {
  it(`OffersSortingItem should render correctly when the type is current`, () => {
    const tree = renderer
      .create(
          <OffersSorting
            toggle={false}
            onOptionClick={noop}
            onToggleClick={noop}
            sortingType={`type1`}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
