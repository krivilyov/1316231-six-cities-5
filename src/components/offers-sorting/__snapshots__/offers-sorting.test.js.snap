// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersSorting should render correctly OffersSortingItem should render correctly when the type is current 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    id="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
  <select
    className="places__sorting-type visually-hidden"
    id="places-sorting"
    onChange={[Function]}
    value="type1"
  >
    <option
      className="places__option popular"
      value="popular"
    >
      Popular
    </option>
    <option
      className="places__option to-high"
      value="to-high"
    >
      Price: low to high
    </option>
    <option
      className="places__option to-low"
      value="to-low"
    >
      Price: high to low
    </option>
    <option
      className="places__option top-rated"
      value="top-rated"
    >
      Top rated first
    </option>
  </select>
</form>
`;
